### PR TITLE
91 create user context

### PR DESCRIPTION
**Pull request by naghamabuwarda**

### CODE REVIEW

File: client/src/context/User.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  setUser?: (user: User) => void,
+};
+export const UserContext = createContext<UserContex>({});
+function UserMemo({ children } : any) {
+  const [user, setUser] = useState<User | null>(null);
+  useEffect(() => {
+    const getUser = async () => {
+      try {
+        const userObj = await axios.get('/api/v1/user');
+        setUser(userObj.data);
```
Comment (LinaYahya) [inline client/src/context/User.tsx:18]:
```suggestion
        const {data} = await axios.get('/api/v1/user');
        setUser(data.data);
```

File: client/src/context/User.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      } catch (err) {
+        console.log(err, 'axios error');
```
Comment (LinaYahya) [inline client/src/context/User.tsx:20]:
```suggestion
setUser(null) // no need to console here
```

File: client/src/context/User.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+} from 'react';
+import axios from 'axios';
+import { User } from '../interfaces';
+
+type UserContex = {
+  user?: User | null,
+  setUser?: (user: User) => void,
+};
+export const UserContext = createContext<UserContex>({});
+function UserMemo({ children } : any) {
```
Comment (LinaYahya) [inline client/src/context/User.tsx:12]:
```suggestion
function UserMemo({ children } : FindTheRealType) {
```

File: client/src/context/User.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import {
+  createContext, useMemo, useState, useEffect,
+} from 'react';
+import axios from 'axios';
+import { User } from '../interfaces';
+
+type UserContex = {
```
Comment (LinaYahya) [inline client/src/context/User.tsx:7]: move to interfaces

File: client/src/context/User.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  user?: User | null,
+  setUser?: (user: User) => void,
+};
+export const UserContext = createContext<UserContex>({});
```
Comment (LinaYahya) [inline client/src/context/User.tsx:11]:
```suggestion
export const UserContext = createContext<UserContext>({});
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by mohammedagha27 — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by shathakh — APPROVED** [review]

**Review by alishanti98 — APPROVED** [review]

**Pull request merged**